# This is the first part of the assignment. 

In [1]:
import numpy as np 
import pandas as pd 
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
import requests
import requests 
print('Libraries imported.')

Libraries imported.


In [3]:

c_wikiPage = requests.get('https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M')
c_data=c_wikiPage.text
x=c_data.find("<table>")
y=c_data.find("</table>")
c_data=c_data[x+6:y]
x=c_data.find("<th>")
y=c_data.find("</th>")
col1=c_data[x+4:y]
c_data=c_data[y+5:]
x=c_data.find("<th>")
y=c_data.find("</th>")
col2=c_data[x+4:y]
c_data=c_data[y+4:]
x=c_data.find("<th>")
y=c_data.find("</th>")
col3=c_data[x+4:y-1]
c_data=c_data[y+2:]
columns_Data=[col1,col2,col3]
CanData=c_data
columns_Data

['Postcode', 'Borough', 'Neighbourhood']

In [4]:
c_data=CanData
CanList=[]
tdExists=True
while tdExists:
    i=0
    Item=[]
    while i<3:
        x=c_data.find("<td>")
        y=c_data.find("</td>")
        if x==-1:
            tdExists=False
        v=c_data[x+4:y]
        if v.find("title") != -1:
            v=v[v.find("title=\""):]
            m=v.find(">")
            n=v.find("</a>")
            v=v[m+1:n]
        v=v.replace("\n","")
        Item.append(v)
        c_data=c_data[y+5:]
        i+=1
    if 'Not assigned' not in str(Item):
        CanList.append(Item)
del CanList[-1]

CanDat=pd.DataFrame(CanList, columns=columns_Data)
CanDat.shape#data shape
CanDat

Postcode           Borough  \
0        M3A        North York   
1        M4A        North York   
2        M5A  Downtown Toronto   
3        M5A  Downtown Toronto   
4        M6A        North York   
5        M6A        North York   
6        M9A         Etobicoke   
7        M1B       Scarborough   
8        M1B       Scarborough   
9        M3B        North York   
10       M4B         East York   
11       M4B         East York   
12       M5B  Downtown Toronto   
13       M5B  Downtown Toronto   
14       M6B        North York   
15       M9B         Etobicoke   
16       M9B         Etobicoke   
17       M9B         Etobicoke   
18       M9B         Etobicoke   
19       M9B         Etobicoke   
20       M1C       Scarborough   
21       M1C       Scarborough   
22       M1C       Scarborough   
23       M3C        North York   
24       M3C        North York   
25       M4C         East York   
26       M5C  Downtown Toronto   
27       M6C              York   
28       M9C         Etobicoke   
29       M9C         Etobicoke   
30       M9C         Etobicoke   
31       M9C         Etobicoke   
32       M1E       Scarborough   
33       M1E       Scarborough   
34       M1E       Scarborough   
35       M4E      East Toronto   
36       M5E  Downtown Toronto   
37       M6E              York   
38       M1G       Scarborough   
39       M4G         East York   
40       M5G  Downtown Toronto   
41       M6G  Downtown Toronto   
42       M1H       Scarborough   
43       M2H        North York   
44       M3H        North York   
45       M3H        North York   
46       M3H        North York   
47       M4H         East York   
48       M5H  Downtown Toronto   
49       M5H  Downtown Toronto   
50       M5H  Downtown Toronto   
51       M6H      West Toronto   
52       M6H      West Toronto   
53       M1J       Scarborough   
54       M2J        North York   
55       M2J        North York   
56       M2J        North York   
57       M3J        North York   
58       M3J        North York   
59       M4J         East York   
60       M5J  Downtown Toronto   
61       M5J  Downtown Toronto   
62       M5J  Downtown Toronto   
63       M6J      West Toronto   
64       M6J      West Toronto   
65       M1K       Scarborough   
66       M1K       Scarborough   
67       M1K       Scarborough   
68       M2K        North York   
69       M3K        North York   
70       M3K        North York   
71       M4K      East Toronto   
72       M4K      East Toronto   
73       M5K  Downtown Toronto   
74       M5K  Downtown Toronto   
75       M6K      West Toronto   
76       M6K      West Toronto   
77       M6K      West Toronto   
78       M1L       Scarborough   
79       M1L       Scarborough   
80       M1L       Scarborough   
81       M2L        North York   
82       M2L        North York   
83       M3L        North York   
84       M4L      East Toronto   
85       M4L      East Toronto   
86       M5L  Downtown Toronto   
87       M5L  Downtown Toronto   
88       M6L        North York   
89       M6L        North York   
90       M6L        North York   
91       M9L        North York   
92       M1M       Scarborough   
93       M1M       Scarborough   
94       M1M       Scarborough   
95       M2M        North York   
96       M2M        North York   
97       M3M        North York   
98       M4M      East Toronto   
99       M5M        North York   
100      M5M        North York   
101      M6M              York   
102      M6M              York   
103      M6M              York   
104      M6M              York   
105      M9M        North York   
106      M9M        North York   
107      M1N       Scarborough   
108      M1N       Scarborough   
109      M2N        North York   
110      M3N        North York   
111      M4N   Central Toronto   
112      M5N   Central Toronto   
113      M6N              York   
114      M6N              York   
115      M9N              York   
116      M1P       Scarbor

In [5]:
CanDat.shape

(211, 3)

# There are 211 rows and 3 columns in the df.